static/js/components/search/CourseFilterDrawer.js

```diff
@@ -28,76 +28,89 @@ type FilterDrawerProps = {
   onUpdateFacets: Function
 }
 
-export function FilterDisplay(props: FilterDrawerProps) {
-  const {
-    activeFacets,
-    clearAllFilters,
-    toggleFacet,
-    facetOptions,
-    onUpdateFacets
-  } = props
+// I had to add a `React.memo` call here because this component
```

Comment:
Just curious, did you figure out what the slow part was? Is there just a lot of content here?

Reply:
I think that's basically it - when we're dealing with the full set of topics and so on it's a fair amount of rendering that's happening. And I think unfortunately it was just sort of all being rendered...

I used the react dev tools and tracked down the component that was taking longest to the `CourseFilterDrawer` iirc

Reply:
but the `React.memo` is pretty smart about how it works, so it will skip unnecessary renders but if there's any call to set state on a subcomponent it will re-render that part of the tree so I think even though it's not clear exactly where the issue is we should be ok. It also wasn't like a terrible, awful slowdown, it just got a bit sluggish - although then again this is on my work laptop which is pretty beefy, so it might have been terrible on a phone.